### PR TITLE
Drop V2: Tabs component

### PR DESCRIPTION
**Pull request by TuvalSimha**
## Summary

- Deleted `Tabs` component from `V2`
- Support 2 types of `Tabs`: basic tabs and menu tabs
- Added new `Tabs` to Storybook - [Link](https://7c1d4283.hive-storybook.pages.dev/?path=/story/components-tabs--default)
- Replace all old Tabs component with the new Tabs.

### CODE REVIEW

File: packages/web/app/src/components/ui/tabs.tsx [hunk 1]
```diff
@@ -6,6 +6,21 @@ import * as TabsPrimitive from '@radix-ui/react-tabs';
 
 const Tabs = TabsPrimitive.Root;
 
+const TabsListMenu = React.forwardRef<
```
Comment (dimaMachina) [inline packages/web/app/src/components/ui/tabs.tsx:9]:
```suggestion
const TabsList = React.forwardRef<
```
can it be similar to radix?

File: packages/web/app/src/components/ui/tabs.tsx [hunk 2]
```diff
@@ -21,14 +36,33 @@ const TabsList = React.forwardRef<
 ));
 TabsList.displayName = TabsPrimitive.List.displayName;
 
+const TabsTriggerMenu = React.forwardRef<
```
Comment (dimaMachina) [inline packages/web/app/src/components/ui/tabs.tsx:39]:
```suggestion
const TabsTrigger = React.forwardRef<
```

File: packages/web/app/src/components/ui/tabs.tsx [hunk 3]
```diff
@@ -21,14 +36,33 @@ const TabsList = React.forwardRef<
...
+  React.ComponentPropsWithoutRef<typeof TabsPrimitive.Trigger> & { hasBorder?: boolean }
+>(({ className, hasBorder = true, ...props }, ref) => (
+  <TabsPrimitive.Trigger
+    ref={ref}
+    className={cn(
+      'cursor-pointer !appearance-none', // unset button styles in Safari
+      'text-sm font-medium text-white transition',
+      hasBorder
+        ? 'radix-state-active:border-b-orange-500 border-b-2 border-b-transparent px-4 py-3 hover:border-b-orange-900'
+        : null,
```
Comment (dimaMachina) [inline packages/web/app/src/components/ui/tabs.tsx:50]:
```suggestion
      hasBorder
        && 'radix-state-active:border-b-orange-500 border-b-2 border-b-transparent px-4 py-3 hover:border-b-orange-900',
```

File: packages/web/app/src/components/ui/tabs.tsx [hunk 4]
```diff
@@ -37,6 +71,18 @@ const TabsTrigger = React.forwardRef<
 ));
 TabsTrigger.displayName = TabsPrimitive.Trigger.displayName;
 
+const TabsContentMenu = React.forwardRef<
```
Comment (dimaMachina) [inline packages/web/app/src/components/ui/tabs.tsx:74]:
```suggestion
const TabsContent = React.forwardRef<
```

File: packages/web/app/src/components/ui/tabs.tsx [hunk 5]
```diff
@@ -37,6 +71,18 @@ const TabsTrigger = React.forwardRef<
...
+  React.ElementRef<typeof TabsPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof TabsPrimitive.Content> & { noPadding?: boolean }
+>(({ className, noPadding, ...props }, ref) => (
+  <TabsPrimitive.Content
+    ref={ref}
+    className={cn(noPadding ? undefined : 'py-7', className)}
```
Comment (dimaMachina) [inline packages/web/app/src/components/ui/tabs.tsx:80]:
```suggestion
    className={cn(!noPadding && 'py-7', className)}
```

File: packages/web/app/src/stories/tabs.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            <a onClick={() => setPage(Page.Policy)}>Policy</a>
+          </TabsTrigger>
+          <TabsTrigger value={Page.Settings} asChild>
+            <a onClick={() => setPage(Page.Settings)}>Settings</a>
+          </TabsTrigger>
+          <TabsTrigger value={Page.Support} asChild>
+            <a onClick={() => setPage(Page.Support)}>Support</a>
+          </TabsTrigger>
+          <TabsTrigger value={Page.Subscription} asChild>
+            <a onClick={() => setPage(Page.Subscription)}>Subscription</a>
```
Comment (dimaMachina) [inline packages/web/app/src/stories/tabs.stories.tsx:44]:
```suggestion
            <button onClick={() => setPage(Page.Members)}>Overview</button>
          </TabsTrigger>
          <TabsTrigger value={Page.Members} asChild>
            <button onClick={() => setPage(Page.Members)}>Members</button>
          </TabsTrigger>
          <TabsTrigger value={Page.Policy} asChild>
            <button onClick={() => setPage(Page.Policy)}>Policy</button>
          </TabsTrigger>
          <TabsTrigger value={Page.Settings} asChild>
            <button onClick={() => setPage(Page.Settings)}>Settings</button>
          </TabsTrigger>
          <TabsTrigger value={Page.Support} asChild>
            <button onClick={() => setPage(Page.Support)}>Support</button>
          </TabsTrigger>
          <TabsTrigger value={Page.Subscription} asChild>
            <button onClick={() => setPage(Page.Subscription)}>Subscription</button>
```

links should have `href` use `button` instead

File: packages/web/app/src/stories/tabs.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            <a onClick={() => setPage(Page.Policy)}>Policy</a>
+          </TabsTriggerMenu>
+          <TabsTriggerMenu value={Page.Settings} asChild>
+            <a onClick={() => setPage(Page.Settings)}>Settings</a>
+          </TabsTriggerMenu>
+          <TabsTriggerMenu value={Page.Support} asChild>
+            <a onClick={() => setPage(Page.Support)}>Support</a>
+          </TabsTriggerMenu>
+          <TabsTriggerMenu value={Page.Subscription} asChild>
+            <a onClick={() => setPage(Page.Subscription)}>Subscription</a>
```
Comment (dimaMachina) [inline packages/web/app/src/stories/tabs.stories.tsx:83]:
```suggestion
            <button onClick={() => setPage(Page.Members)}>Overview</button>
          </TabsTriggerMenu>
          <TabsTriggerMenu value={Page.Members} asChild>
            <button onClick={() => setPage(Page.Members)}>Members</button>
          </TabsTriggerMenu>
          <TabsTriggerMenu value={Page.Policy} asChild>
            <button onClick={() => setPage(Page.Policy)}>Policy</button>
          </TabsTriggerMenu>
          <TabsTriggerMenu value={Page.Settings} asChild>
            <button onClick={() => setPage(Page.Settings)}>Settings</button>
          </TabsTriggerMenu>
          <TabsTriggerMenu value={Page.Support} asChild>
            <button onClick={() => setPage(Page.Support)}>Support</button>
          </TabsTriggerMenu>
          <TabsTriggerMenu value={Page.Subscription} asChild>
            <button onClick={() => setPage(Page.Subscription)}>Subscription</button>
```

File: packages/web/app/src/stories/tabs.stories.tsx (hunk 6 above)
Comment (dimaMachina) [reply packages/web/app/src/stories/tabs.stories.tsx:44]: also use `.map(...)` to DRY

File: packages/web/app/src/components/ui/tabs.tsx (hunk 5 above)
Comment (dimaMachina) [reply packages/web/app/src/components/ui/tabs.tsx:80]: personally, I don't like I don't like `props` aka `noPadding`, `noMargin`, you have already `className` and padding can be removed with `p-0` cc @kamilkisiela @n1ru4l

File: packages/web/app/src/components/ui/tabs.tsx (hunk 1 above)
Comment (dimaMachina) [reply packages/web/app/src/components/ui/tabs.tsx:9]: maybe use `variant="..."` instead since there is 2 types of tabs

**Review by dimaMachina — CHANGES_REQUESTED** [review]
overview is not clickable

https://github.com/user-attachments/assets/aff26079-9e38-42c7-aad2-86da69e5b933

**Comment by TuvalSimha** [comment]
@dimaMachina 

I changed it to use `cva` and `variants`, this looks much better now. What do you think?

File: packages/web/app/src/components/ui/tabs.tsx [hunk 8]
```diff
@@ -1,50 +1,89 @@
...
->(({ className, ...props }, ref) => (
+  React.ComponentPropsWithoutRef<typeof TabsPrimitive.Content> &
+    VariantProps<typeof tabsContentVariants> & { noPadding?: boolean }
+>(({ className, variant, noPadding, ...props }, ref) => (
   <TabsPrimitive.Content
     ref={ref}
     className={cn(
-      'ring-offset-background focus-visible:ring-ring mt-2 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2',
+      tabsContentVariants({ variant }),
+      !noPadding && tabsContentVariants({ variant: 'default' }),
```
[the remark this answers is not in the document]
Comment (TuvalSimha) [reply packages/web/app/src/components/ui/tabs.tsx]: Done :)

**Review by saihaj — APPROVED** [review]

**Review by saihaj — CHANGES_REQUESTED** [review]
Hmm actually @dimaMachina comment around not being able to click `overview` on storybook is stil valid.

On the actual app usage I don't see this issue. So something weird going on with storybook 😢 

https://github.com/user-attachments/assets/f506c5c2-0d68-4b0c-ab72-980a0fc0da6c

**Review by saihaj — APPROVED** [review]
Actually you are good. The link in the PR description is pointing to old preview. If I test latest storybook build it works as expected.

**Pull request merged**